src/storage/src/hummock/sstable/block.rs

```diff
@@ -569,7 +660,40 @@ impl BlockBuilder {
         self.buf
             .put_u32_le(self.restart_points_type_index.len() as u32);
 
+        let mut epoch_index: usize = 0;
+        let epoch_dictionary = self
+            .epoch_dictionary
+            .iter()
+            .cloned()
+            .map(|epoch| {
+                epoch_index += 1;
+                (epoch, epoch_index)
+            })
+            .collect::<BTreeMap<HummockEpoch, usize>>();
+
+        // encode epoch_group
+        for group in &self.epoch_group {
+            let group_len = group.len();
+            self.buf.put_u16_le(group_len as u16);
```

Comment:
Although highly unlikely, the possibility of group_len overflow still exists. How about focing a block swtich when the number of keys added to the block >= u16::MAX?